### PR TITLE
refactor: Change C style casts to C++ style (Part 7)

### PR DESCRIPTION
**Pull request by aditi-pandit**
As per the security guideline in
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Findings in many miscellaneous files.

### CODE REVIEW

File: velox/external/date/tz.cpp [hunk 1]
```diff
@@ -2834,7 +2834,7 @@ std::unique_ptr<CURL, curl_deleter>
 curl_init()
 {
     static const auto curl_is_now_initiailized = curl_global();
-    (void)curl_is_now_initiailized;
+    static_casT<void>(curl_is_now_initiailized);
```
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/external/date/tz.cpp]: Done.

File: velox/parse/Expressions.cpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ std::vector<TypePtr> implicitCastTargets(const TypePtr& type) {
       break;
     }
     default: // make compilers happy
-      (void)0; // Statement to avoid empty semicolon warning
+      static_cast<void>(0); // Statement to avoid empty semicolon warning
```
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/parse/Expressions.cpp]: I don't think (void)0 is worthwhile either, but was just limiting to fixing the casts. break is cleaner.

**Comment by kgpai** [comment]
@aditi-pandit can you fix the conflict before I import ?

**Comment by aditi-pandit** [comment]
> @aditi-pandit can you fix the conflict before I import ?

@kgpai : Hi Krishna, Have rebased the code. Please let me know if you need anything else.

**Comment by kgpai** [comment]
@aditi-pandit Can you rebase again ? Certain builds/tests were broken in main.

**Comment by aditi-pandit** [comment]
Sure @kgpai. Have rebased and pushed again.

**Pull request closed without merging**